brickstrap-components.sh

```diff
@@ -137,21 +258,46 @@ function brp_validate_component_names()
 # arguments passed to this function are passed along to the callback aftet the
 # component name.
 #
+# $1: the list of components to iterate over.
+# $2: the callback to invoke.
+# $3...: optional: additional arguments passed to the callback following the
+#                  component name.
+#
+function brp_iterate_components_impl()
+{
+    # note: $2 may be empty, which is valid.
+    if [ $# -lt 3 -o -z "$1" -o -z "$3" ]; then
```

Comment:
Looks like the hang-up is here. This fails if there are less than 3 arguments, but arg 3 is optional and `brp_iterate_components_impl` is called with 2 arguments from line 300.